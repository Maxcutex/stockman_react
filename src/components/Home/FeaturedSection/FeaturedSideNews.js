import React from 'react';
import FeaturedSideNewsCaption from './FeaturedSideNewsCaption';
import FeaturedSideNewsTagList from './FeaturedSideNewsTagList';

export default props => {
    const { title, visual_news, author,category_news } = props.featured;
    console.log(props.featured);
    // const visual_news_main = visual_news.filter(
    //     image => image.image_type === 'size450x330'
    // );
    return (
        <div className="post-type-1">
            {visual_news.map((vn, i) => {
                return (
                    vn.image_type === 'size450x330' && (
                        <img
                            src={vn.image_file}
                            width="450"
                            height="330"
                            alt=""
                            key={i}
                        />
                    )
                );
            })}
            <FeaturedSideNewsTagList category_news={category_news} />
            <FeaturedSideNewsCaption title={title} author={author} />
        </div>
    );
};
