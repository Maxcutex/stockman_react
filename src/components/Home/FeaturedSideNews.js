import React from "react";

export default () => {
  return (
    <div className="post-type-1">
      <img src="images/home-1-450x330.jpg" width="450" height="330" alt="" />
      <ul className="tag-list">
        <li>
          <a href="post.html">Money</a>
        </li>
        <li>
          <a href="post.html">Life</a>
        </li>
      </ul>
      <div className="caption">
        <h4 className="title">
          <a href="post.html">
            Dollar and bond yields rise as US growth hits 2-year high
          </a>
        </h4>
        <div className="bottom-block">
          <ul className="meta-list">
            <li>
              by <a href="post.html">Jon Snow</a>
            </li>
            <li>
              <a href="post.html">Aug 08, 2017 </a>
            </li>
          </ul>
          <div className="share-block">
            <div className="soc-icon">
              <a className="icon fa-twitter" href="/" />
              <a className="icon fa-facebook-square" href="/" />
              <a className="icon fa-instagram" href="/" />
              <a className="icon fa-pinterest" href="/" />
              <a className="icon fa-youtube-play" href="/" />
            </div>
            <span className="icon fa-share-alt" />
          </div>
        </div>
      </div>
    </div>
  );
};
