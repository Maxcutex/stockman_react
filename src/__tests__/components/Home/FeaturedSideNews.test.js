import React from 'react';
import { shallow } from 'enzyme';

import FeaturedSideNews from '../../../components/Home/FeaturedSideNews';
describe('FeaturedSideNews Component', () => {
    let component;
    let props;
    beforeEach(() => {
        props = {
            featured: {
                visual_news: [
                    {
                        image_file: 'abi.jpg',
                        image_type: 'size450x330',
                    },
                    {
                        image_file: 'abi.jpg',
                        image_type: 'size950x330',
                    },
                ],
                title: 'test title',
                author: 'test',
                id: 1,
            },
        };
        component = shallow(
            <FeaturedSideNews {...props} key={props.featured.id} />
        );
    });
    it('renders without crashing', () => {
        expect(component).toMatchSnapshot();
    });
});
