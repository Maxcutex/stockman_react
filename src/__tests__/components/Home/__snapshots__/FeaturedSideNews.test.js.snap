// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedSideNews Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    featured={
      Object {
        "author": "test",
        "id": 1,
        "title": "test title",
        "visual_news": Array [
          Object {
            "image_file": "abi.jpg",
            "image_type": "size450x330",
          },
          Object {
            "image_file": "abi.jpg",
            "image_type": "size950x330",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <img
            alt=""
            height="330"
            src="abi.jpg"
            width="450"
          />,
          false,
        ],
        <FeaturedSideNewsTagList />,
        <FeaturedSideNewsCaption
          author="test"
          title="test title"
        />,
      ],
      "className": "post-type-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "height": "330",
          "src": "abi.jpg",
          "width": "450",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": "test",
          "title": "test title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <img
              alt=""
              height="330"
              src="abi.jpg"
              width="450"
            />,
            false,
          ],
          <FeaturedSideNewsTagList />,
          <FeaturedSideNewsCaption
            author="test"
            title="test title"
          />,
        ],
        "className": "post-type-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "height": "330",
            "src": "abi.jpg",
            "width": "450",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": "test",
            "title": "test title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
